app/main.py

```diff
@@ -2,0 +1,18 @@
+from app.cafe import Cafe
+from app.errors import NotWearingMaskError, VaccineError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_needed = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_needed += 1
+
+    if masks_needed > 0:
+        return f"Friends should buy {masks_needed} masks"
```

Comment:
Code Efficiency: There's no need to compare `masks_needed` to 0. You can just check if `masks_needed` is truthy. So, instead of `if masks_needed > 0:`, use `if masks_needed:`.